trunk/source/Library-CHC/src/de/uni_freiburg/informatik/ultimate/lib/chc/IChcScript.java

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright (C) 2023 Dominik Klumpp (klumpp@informatik.uni-freiburg.de)
+ * Copyright (C) 2023 University of Freiburg
+ *
+ * This file is part of the ULTIMATE CHC Library.
+ *
+ * The ULTIMATE CHC Library is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * The ULTIMATE CHC Library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with the ULTIMATE CHC Library. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Additional permission under GNU GPL version 3 section 7:
+ * If you modify the ULTIMATE CHC Library, or any covered work, by linking
+ * or combining it with Eclipse RCP (or a modified version of Eclipse RCP),
+ * containing parts covered by the terms of the Eclipse Public License, the
+ * licensors of the ULTIMATE CHC Library grant you additional permission
+ * to convey the resulting work.
+ */
+package de.uni_freiburg.informatik.ultimate.lib.chc;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+import de.uni_freiburg.informatik.ultimate.logic.Model;
+import de.uni_freiburg.informatik.ultimate.logic.Script;
+import de.uni_freiburg.informatik.ultimate.logic.Script.LBool;
+
+/**
+ * A common interface to CHC solvers.
+ *
+ * @author Dominik Klumpp (klumpp@informatik.uni-freiburg.de)
+ *
+ */
+public interface IChcScript {
+	/**
+	 * The script which should be used to build terms for this solver.
+	 *
+	 * No other commands should be invoked on this script.
+	 *
+	 * @return
+	 */
+	Script getScript();
+
+	/**
+	 * Try to solve the given system of Horn clauses.
+	 *
+	 * The instance may have an internal timeout, after which {@code UNKNOWN} is returned.
+	 *
+	 * @param symbolTable
+	 *            A symbol table describing the predicates and other symbols used by the given CHC system
+	 * @param system
+	 *            The list of Horn clauses forming the CHC system
+	 * @return the satisfiability of the system
+	 */
+	LBool solve(HcSymbolTable symbolTable, List<HornClause> system);
+
+	/**
+	 * Try to solve the given system of Horn clauses, within the given time.
+	 *
+	 * @param symbolTable
+	 *            A symbol table describing the predicates and other symbols used by the given CHC system
+	 * @param system
+	 *            The list of Horn clauses forming the CHC system
+	 * @param timeout
```

Comment:
what is the valid range of the timeout? what happens with timeout=0? what happens if the timeout is reached?

Reply:
```
* @param timeout
*            The maximum time to be used for solving the system, in milliseconds. The timeout must be greater than
*            zero. If the timeout is reached before a solution is found, {@code UNKNOWN} is returned.
```